### DISCUSSION TITLE
is there a list of all the lock free oriented cpu intrinsics ?

### DISCUSSION THREAD

**Discussion by ymolists** [dendibakh/perf-book#53] [Q&A]
Hi Folks

I remember back in 2014 lock free was the rage with intel/arm cpu inrinsics[1]. Are there any new cpu instructions that came since then ? i feel no one is talking about any new improvments that happened since then.

this book could maybe have one chapter or a link to resources outlining those instructions :-) 

1) for example LOCK XADD and XCHG on intel

**Comment by dendibakh** [comment]
I haven't looked into it, I don't know.
